### DISCUSSION TITLE
Interpolate from array in different positions of the translation string

### DISCUSSION THREAD

**Discussion by olrtg** [i18next/i18next#1789] [Q&A] [answered]
Hey guys! Currently working at a notification module for work. I have an API that sends me the type of notification so from a table i select dynamically the translation that I'm going to use.

So for example, if the notification type is `surveyStoplightNoteAddded`, I'm going to use `t('notifications.codes.surveyStoplightNoteAdded')` and so on.
```js
const NOTIFICATION_CODES = {
  surveyStoplightNoteAdded: i18n.t(
    'notifications.codes.surveyStoplightNoteAdded',
  ),
}
```

Also I'm getting an array of strings that i need to interpolate inside the selected translation. Something like this:
```json
{
  "notifications": {
    "codes": {
      "surveyStoplightNoteAdded": "{createdBy} has left a comment on indicator {0} on survey {1}"
    }
  }
}
```
But depending of the notification type the amount of interpolations may vary. I've found how to make interpolation using `{{key}}` and then passing an object as second argument, but that's only helpful if I have a fixed amount of keys to interpolate, so I'm wondering how can I implement this.
Any ideas?

**Comment by jamuhl** [comment, marked as answer]
> but that's only helpful if I have a fixed amount of keys to interpolate

not really ... the call is `t('notifications.codes.surveyStoplightNoteAdded', options)` -> the options can contain values to be interpolated

can even be some nested object: https://www.i18next.com/translation-function/interpolation#working-with-data-models

**Reply by olrtg** [reply]
Hey thank you so much for the quick reply! Following your recommendations what I did was build the `options` object dynamically given the array that comes from the API (converted from array of strings to object `{ createdBy: 'admin', 0: 'Some value', 1: 'Another value'}` ).
